src/comm_droplet.F

```diff
@@ -686,35 +853,84 @@ subroutine setupDepartDroplet(Depart_ind,ptrDepart,pdata_neighbor)
       sw_idx = ptrDepart(isw)
       se_idx = ptrDepart(ise)
       do np=1, nparcelsLocalActive
-        if(pdata_neighbor(np) .eq. mynorth) then
+        if(pdata_neighbor(np) .eq. inorth) then
           Depart_ind(n_idx) = np
           n_idx=n_idx+1
-        else if(pdata_neighbor(np) .eq. mysouth) then
+        else if(pdata_neighbor(np) .eq. isouth) then
           Depart_ind(s_idx) = np
           s_idx=s_idx+1
-        else if(pdata_neighbor(np) .eq. mywest) then
+        else if(pdata_neighbor(np) .eq. iwest) then
           Depart_ind(w_idx) = np
           w_idx=w_idx+1
-        else if(pdata_neighbor(np) .eq. myeast) then
+        else if(pdata_neighbor(np) .eq. ieast) then
           Depart_ind(e_idx) = np
           e_idx=e_idx+1
-        else if(pdata_neighbor(np) .eq. mynw) then
+        else if(pdata_neighbor(np) .eq. inw) then
           Depart_ind(nw_idx) = np
           nw_idx=nw_idx+1
-        else if(pdata_neighbor(np) .eq. myne) then
+        else if(pdata_neighbor(np) .eq. ine) then
           Depart_ind(ne_idx) = np
           ne_idx=ne_idx+1
-        else if(pdata_neighbor(np) .eq. mysw) then
+        else if(pdata_neighbor(np) .eq. isw) then
           Depart_ind(sw_idx) = np
           sw_idx=sw_idx+1
-        else if(pdata_neighbor(np) .eq. myse) then
+        else if(pdata_neighbor(np) .eq. ise) then
           Depart_ind(se_idx) = np
           se_idx=se_idx+1
         endif
       enddo
+      !$acc update device(Depart_ind)
+#endif
+
+ 110     format('myid: ',i4,' ',a10,' ',8(i8,1x))
 
      end subroutine setupDepartDroplet
 
+#if (defined(_COUNTPREFIX) && defined(_OPENACC))
+!ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
+!ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
+!ccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc
+
+     subroutine GatherIndices(dir,ptrDepart,Cidx,pdata_neighbor,Depart_ind)
```

Comment:
Where is this subroutine being called?

Reply:
If you want to figure out why this subroutine is failing at runtime, then we could replace a large amount of the duplicate code in setupDepartDroplet.